cmd/server/pactasrv/analysis.go

```diff
@@ -0,0 +1,343 @@
+	"github.com/RMI/pacta/db"
+	"github.com/RMI/pacta/oapierr"
+	api "github.com/RMI/pacta/openapi/pacta"
+	"github.com/RMI/pacta/pacta"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// (GET /analyses)
+func (s *Server) ListAnalyses(ctx context.Context, request api.ListAnalysesRequestObject) (api.ListAnalysesResponseObject, error) {
+	ownerID, err := s.getUserOwnerID(ctx)
+	if err != nil {
+		return nil, err
+	}
+	as, err := s.DB.AnalysesByOwner(s.DB.NoTxn(ctx), ownerID)
+	if err != nil {
+		return nil, oapierr.Internal("failed to query analyses", zap.Error(err))
+	}
+	items, err := dereference(conv.AnalysesToOAPI(as))
+	if err != nil {
+		return nil, err
+	}
+	return api.ListAnalyses200JSONResponse{Items: items}, nil
+}
+
+// Deletes an analysis (and its artifacts) by ID
+// (DELETE /analysis/{id})
+func (s *Server) DeleteAnalysis(ctx context.Context, request api.DeleteAnalysisRequestObject) (api.DeleteAnalysisResponseObject, error) {
+	id := pacta.AnalysisID(request.Id)
+	_, err := s.checkAnalysisAuthorization(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	blobURIs, err := s.DB.DeleteAnalysis(s.DB.NoTxn(ctx), id)
+	if err != nil {
+		return nil, oapierr.Internal("failed to delete analysis", zap.Error(err))
+	}
+	if err := s.deleteBlobs(ctx, blobURIs...); err != nil {
+		return nil, err
+	}
+	return api.DeleteAnalysis204Response{}, nil
+}
+
+// Returns an analysis by ID
+// (GET /analysis/{id})
+func (s *Server) FindAnalysisById(ctx context.Context, request api.FindAnalysisByIdRequestObject) (api.FindAnalysisByIdResponseObject, error) {
+	a, err := s.checkAnalysisAuthorization(ctx, pacta.AnalysisID(request.Id))
+	if err != nil {
+		return nil, err
+	}
+	if err := s.populateArtifactsInAnalyses(ctx, a); err != nil {
+		return nil, err
+	}
+	if err := s.populateBlobsInAnalysisArtifacts(ctx, a.Artifacts...); err != nil {
+		return nil, err
+	}
+	converted, err := conv.AnalysisToOAPI(a)
+	if err != nil {
+		return nil, err
+	}
+	return api.FindAnalysisById200JSONResponse(*converted), nil
+}
+
+// Updates writable analysis properties
+// (PATCH /analysis/{id})
+func (s *Server) UpdateAnalysis(ctx context.Context, request api.UpdateAnalysisRequestObject) (api.UpdateAnalysisResponseObject, error) {
+	id := pacta.AnalysisID(request.Id)
+	_, err := s.checkAnalysisAuthorization(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	mutations := []db.UpdateAnalysisFn{}
+	if request.Body.Name != nil {
+		mutations = append(mutations, db.SetAnalysisName(*request.Body.Name))
+	}
+	if request.Body.Description != nil {
+		mutations = append(mutations, db.SetAnalysisDescription(*request.Body.Description))
+	}
+	err = s.DB.UpdateAnalysis(s.DB.NoTxn(ctx), id, mutations...)
+	if err != nil {
+		return nil, oapierr.Internal("failed to update analysis", zap.Error(err))
+	}
+	return api.UpdateAnalysis204Response{}, nil
+}
+
+func (s *Server) checkAnalysisAuthorization(ctx context.Context, id pacta.AnalysisID) (*pacta.Analysis, error) {
```

Comment:
Strong +1 to the more explicit authorization approach, it's easy to understand + modify, with little room for sneaky (performance + security) footguns

Reply:
Generally agree - I think we can handle other cases as they arise. The more I work with this project, the more I like the simplicity of its data model in making Authz trivial. That may change as we get into more advanced listing mechanisms, however.